docs/async/async-in-depth.md

```diff
@@ -0,0 +1,121 @@
+# Async In Depth
+
+By [Phillip Carter](https://github.com/cartermp)
+
+Writing I/O-Bound or CPU-Bound asynchronous code is simple using the `async` and `await` keywords, but what's happening under the covers is fairly complex.  The two key objects involved are `Task` and `Task<T>`.
+
+## Task and Task<T>
+
+Tasks are constructs used to represent operations running in the background.
+
+*   `Task` represents a single operation which does not return a value.
+*   `Task<T>` represents a single operation which returns a value of type `T`.
+
+Tasks are awaitable, meaning that using `await` will allow your application or service to perform useful work while the task is running by yielding control to its caller until the task is done. If you’re using `Task<T>`, the `await` keyword will additionally “unwrap” the value returned when the Task is complete.
```

Comment:
There is more to it that stated. I think its important to describe tasks before getting to await. For example, sometimes I await a task on the same line and then other times, I store the task and await it later, either individually or within a group of Tasks (Task.Whenall). Tasks don't have any specific affinity to await. await is a particular (and most popular) client.